### PR TITLE
Implements/126 - Describe how authentication with the web-api works

### LINKED ISSUE #126: Document authentication model
Generally how the [OAuth2](https://oauth.net/2/) model with [JWT](https://tools.ietf.org/html/rfc7523) tokens work, as well as how it is implemented in GIRAF. Should include a diagram that helps with visualisation. Could also include how to do this via `curl`, e.g.

```curl
curl -X POST -d '{"username": <user>, "password": <pw>}' -
H "Content-Type: application/json"
http://localhost:5000/v1/account/login

curl -H 'Accept: application/json' -H "Authorization: Bearer
${TOKEN}" https://localhost:5000/v1/pictogram
```

### PR DESCRIPTION
**Pull request by rasmus91**
This PR adds information to the docs/development/rest_api_development/TokenAuthentication.md file about how JWT works, how it is used in Giraf and how to use it, with curl samples.

The file renders a bit oddly in git, because it uses a flex to show images side by side (I would have used a md table, but mkdocs does not allow images in those) but it renders like this:
Skærmbillede fra 2020-05-15 12-16-11

when mkdocs is serving it.

The pictures are clickable, which will result in a new tab opening with the full size image.

### CODE REVIEW

**Review by mads-bo — COMMENTED** [review]
@aau-giraf/sw608f20 you have been assigned this. Here's your checklist.

Please use the following checklist:

**Code Documentation Only**

- [ ] It is documented which BLoCs/Screens the current class is related to
- [ ] The responsibility/functionality of the current class is explained. Also regarding the classes it is related to
- [ ] If any unusual decisions are introduced in the code, they reasoned about in the documentation
 
**Wiki Related**

- [ ] There are no dead links or references
- [ ] The markdown renders correctly in MkDocs
- [ ] The content is placed in the correct page/folder/section

You can run MkDocs locally by following [this guide](https://aau-giraf.github.io/wiki/writing_wiki/#running-locally)

**Review by mads-bo — COMMENTED** [review]
@aau-giraf/sw617f20 you have been assigned this. Here's your checklist.

Please use the following checklist:

**Code Documentation Only**

- [ ] It is documented which BLoCs/Screens the current class is related to
- [ ] The responsibility/functionality of the current class is explained. Also regarding the classes it is related to
- [ ] If any unusual decisions are introduced in the code, they reasoned about in the documentation
 
**Wiki Related**

- [ ] There are no dead links or references
- [ ] The markdown renders correctly in MkDocs
- [ ] The content is placed in the correct page/folder/section

You can run MkDocs locally by following [this guide](https://aau-giraf.github.io/wiki/writing_wiki/#running-locally)

File: docs/development/rest_api_development/TokenAuthentication.md [hunk 1]
```diff
@@ -11,4 +14,92 @@ This login request will return a token.
...
+        "JwtExpireDays": 30
+    }
+    ...
+}
+```
+
+
+## What is a JWT?
+
+JSON Web Token is an authentication token, which actually contains user information, as opposed to API keys.Traditionally api keys can be a GUID or a nonsense string, which is really a primarykey in a database, when supplied in the header of an HTTP request, the backend will then make an appropriate lookup across tables or with its authentication service to ascertain the rights of the user in question, if any. A JWT works differently: It actually contains serialized information about the user, and their access rights, this has the advantage of saving the service from database lookups, to obtain the information. It also contains a timstamp, by which the service can determine whether it is still valid.
```
Comment (elaurs) [inline docs/development/rest_api_development/TokenAuthentication.md]:
```suggestion
JSON Web Token is an authentication token, which actually contains user information, as opposed to API keys. Traditionally API keys can be a GUID or a nonsense string, which is really a primary key in a database, when supplied in the header of an HTTP request, the backend will then make an appropriate lookup across tables or with its authentication service to ascertain the rights of the user in question, if any. A JWT works differently: It actually contains serialized information about the user, and their access rights, this has the advantage of saving the service from database lookups, to obtain the information. It also contains a timestamp, by which the service can determine whether it is still valid.
```

Comment (elaurs) [reply docs/development/rest_api_development/TokenAuthentication.md]: It is just two added spaces and two spelling mistakes fixed.

File: docs/development/rest_api_development/TokenAuthentication.md [hunk 2]
```diff
@@ -11,4 +14,92 @@ This login request will return a token.
...
+        "JwtExpireDays": 30
+    }
+    ...
+}
+```
+
+
+## What is a JWT?
+
+JSON Web Token is an authentication token, which actually contains user information, as opposed to API keys. Traditionally api keys can be a GUID or a nonsense string, which is really a primary key in a database, when supplied in the header of an HTTP request, the backend will then make an appropriate lookup across tables or with its authentication service to ascertain the rights of the user in question, if any. A JWT works differently: It actually contains serialized information about the user, and their access rights, this has the advantage of saving the service from database lookups, to obtain the information. It also contains a timestamp, by which the service can determine whether it is still valid.
```
Comment (elaurs) [inline docs/development/rest_api_development/TokenAuthentication.md]:
```suggestion
JSON Web Token is an authentication token, which actually contains user information, as opposed to API keys. Traditionally, API keys can be a GUID or a nonsense string, which is really a primary key in a database, when supplied in the header of an HTTP request, the backend will then make an appropriate lookup across tables or with its authentication service to ascertain the rights of the user in question, if any. A JWT works differently: It actually contains serialized information about the user, and their access rights, this has the advantage of saving the service from database lookups, to obtain the information. It also contains a timestamp, by which the service can determine whether it is still valid.
```

Comment (elaurs) [reply docs/development/rest_api_development/TokenAuthentication.md]: Correction: "api" -> "API"

**Review by elaurs — APPROVED** [review]

**Review by Semicolon1071 — APPROVED** [review]

**Comment by mathiasandresen** [comment]
@rasmus91 husk at slette branchen

**Pull request merged**